src/zoid/qr-code/component.jsx

```diff
@@ -143,6 +147,12 @@ export function QRCodeContainer({
         frame.classList.remove(CLASS.INVISIBLE);
         frame.classList.add(CLASS.VISIBLE);
 
+        // Close is now controlled by QRCard on smart-payment-buttons
+        const closeButton = document.querySelector('#close');
+        if (closeButton) {
```

Comment:
This is gonna be true 100% of the time right? Why not just remove the close button from this template entirely and just not render it in the first place?

Reply:
![image](https://user-images.githubusercontent.com/33497086/134065929-4e33f2f7-d06d-425c-94c4-486d03ecac4b.png)
@bluepnume 
While we are on the loading state the iframe component is not ready to be displayed. We need someway to close the modal during that time or even if we have an error on loading the child component.

Reply:
Sure, then can we put the close button in the prerender template? The prerender will be automatically removed when the full component is rendered.

`document.querySelector` isn't very safe here, as you could end up removing elements from the merchant page this way if they have the same ID

Reply:
Sure, know that you are saying, it makes complete sense to not use `document.querySelector`.
I guess moving it to prerender template is a great idea!